src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestStream.cs

```diff
@@ -100,7 +100,7 @@ public override void Write(byte[] buffer, int offset, int count)
                 throw new ArgumentOutOfRangeException("count");
             }
 
-            if (offset + count > buffer.Length)
+            if ((long)offset + (long)count > (long)buffer.Length)
```

Comment:
If we can solve it without casts by reordering the math, then I'm fine.

But while you are changing this, you should also change the corresponding HttpResponseStream.cs file as well and change those casts into the same new math.   As long as we have proper unit tests, we're good.